query.go

```diff
@@ -375,6 +376,9 @@ func (c *Conn) Query(sql string) (*Rows, error) {
 		if _, ok := err.(mysqlproto.ERRPacket); !ok {
 			c.valid = false
 		}
+		if err, ok := err.(net.Error); ok && err.Timeout() {
```

Comment:
I found next version of error handling is more structured:

```
    if err != nil {
        switch err {
        case errNotFound:
            log.Println("Requested item not found")
        default:
            log.Println("Unknown error occurred")
        }
    }
```

Here is article about it:  https://medium.com/@sebdah/go-best-practices-error-handling-2d15e1f0c5ee e.g:

Reply:
The first condition checks that if the error is NOT `mysqlproto.ERRPacket` then we mark the connection as `valid=false` and throw it away (as the exception is not the one we expect). It will also cover the case for `err.(net.Error)`.

Reply:
Ah, good point. I misread the first condition. I'll drop this branch as it's redundant.